mbed_devices/_internal/windows/component_descriptor.py

```diff
@@ -0,0 +1,123 @@
+"""Defines a generic Win32 component."""
+import logging
+from abc import ABC, abstractmethod
+from typing import List, Any, Generator, Optional
+
+import win32com.client
+
+UNKNOWN_VALUE = "Unknown"
+NAMED_TUPLE_FIELDS_ATTRIBUTE = "_fields"
+
+logger = logging.getLogger(__name__)
+
+
+class ComponentDescriptor(ABC):
+    """Win32 component descriptor."""
+
+    @property
+    @abstractmethod
+    def class_name(self) -> str:
+        """Returns the name of the Win32 Class."""
+        return UNKNOWN_VALUE
+
+    @property
+    def parameters(self) -> List[str]:
+        """Returns all the different parameters of the class."""
+        return self._class_parameters()
+
+    @classmethod
+    def _class_parameters(cls) -> List[str]:
+        return [k for k in getattr(cls, NAMED_TUPLE_FIELDS_ATTRIBUTE)]
+
+    @property
+    @abstractmethod
+    def device_id(self) -> str:
+        """Returns the device ID field."""
+        pass
+
+    @property
+    def filter(self) -> Optional[str]:
+        """Filter applied on the list of elements."""
+        return None
+
+    @property
+    def is_undefined(self) -> bool:
```

Comment:
This is a hidden `None` check. I think we should go with pattern similar to `Candidate`, where _empty_ instances should not event exist. Otherwise the burden is on whoever uses the instance to check if it's empty or not. Otherwise, what's the use of `empty` value?